src/users/users.controller.ts

```diff
@@ -72,11 +75,56 @@ export class UsersController {
     return this.usersService.update(id, updateUserDto);
   }
 
+  @Put('username/:username')
+  @Roles([UserRoleEnum.ADMIN, UserRoleEnum.USER])
+  updateByUsername(
+    @Param('username') username: string,
+    @Body() updateUserDto: UpdateUserDto,
+  ): Promise<UserModel> {
+    const message = `UsersController.updateByUsername() username=${username} updateUserDto=${JSON.stringify(
+      updateUserDto,
+    )}`;
+    this.logger.log(message);
+    return this.usersService.updateByUsername(username, updateUserDto);
+  }
+
   @Delete(':id')
   @Roles([UserRoleEnum.ADMIN, UserRoleEnum.USER])
   delete(@Param('id', ParseUUIDPipe) id: string): Promise<number> {
     const message = `UsersController.delete() id=${id}`;
     this.logger.log(message);
     return this.usersService.delete(id);
   }
+
+  //partner endpoints
+  @Post('/partner/add')
+  @Roles([UserRoleEnum.ADMIN, UserRoleEnum.USER])
+  async sendPartnerRequest(@Request() req, @Query('target') target: string) {
+    const message = `partnerController.sendPartnerRequest() req.user=${JSON.stringify(
+      req.user,
+    )}`;
+    this.logger.log(message);
+    return await this.usersService.addPartner(req.user.username, target);
+  }
```

Comment:
let's remove `async` and `await` here and other similar places if you notice. i think [this post](https://stackoverflow.com/questions/52955214/best-practices-for-return-async-promise-typescript) is helpful for the reasons why removing them.